moveit_core/planning_request_adapter/include/moveit/planning_request_adapter/planning_request_adapter.h

```diff
@@ -103,10 +89,11 @@ class PlanningRequestAdapter
      current state itself appears to touch obstacles). This is helpful because the added states should not be considered
      invalid in all situations.
    *  @return True if response got generated correctly */
-  bool adaptAndPlan(const planning_interface::PlannerManagerPtr& planner,
-                    const planning_scene::PlanningSceneConstPtr& planning_scene,
-                    const planning_interface::MotionPlanRequest& req, planning_interface::MotionPlanResponse& res,
-                    std::vector<std::size_t>& added_path_index) const;
+  [[nodiscard]] bool adaptAndPlan(const planning_interface::PlannerManagerPtr& planner,
+                                  const planning_scene::PlanningSceneConstPtr& planning_scene,
+                                  const planning_interface::MotionPlanRequest& req,
+                                  planning_interface::MotionPlanResponse& res,
+                                  std::vector<std::size_t>& added_path_index = EMPTY_PATH_INDEX_VECTOR) const;
```

Comment:
This pattern of passing the default `std::vector<std::size_t>& added_path_index = EMPTY_PATH_INDEX_VECTOR` isn't making sense to me. It is an out-parameter? If the user doesn't pass their own, does it not mutate the global `EMPTY_PATH_INDEX_VECTOR`, which surely is not the intent?

Reply:
Also note that clang-tidy CI is complaining about the naming of this and failing.

Reply:
My intend with this was to remove these functions:

```cpp
bool PlanningRequestAdapterChain::adaptAndPlan(const planning_interface::PlannerManagerPtr& planner,
                                               const planning_scene::PlanningSceneConstPtr& planning_scene,
                                               const planning_interface::MotionPlanRequest& req,
                                               planning_interface::MotionPlanResponse& res) const
{
  std::vector<std::size_t> empty_added_path_index;
  return adaptAndPlan(planner, planning_scene, req, res, empty_added_path_index);
}
```
As far as I understand there are no optional references, so like before we just write the path indices to a memory location we never read and dump them that way. Maybe another possible implementation would be to change this interface to use a pointer but that would be a bigger change. Do you have a smarter idea of how to do this?

Reply:
can't you just use `std::vector<std::size_t>& added_path_index = {}`?

Reply:
@henningkayser C++ won't let you assign a default reference like that.

@sjahr I guess this pattern will technically work as long as `adaptAndPlan` makes sure to clear the "empty" vector before it starts every time. Otherwise it's a memory leak!

Reply:
oh of course not... content-wise, this parameter would fit into the MotionPlanResponse as well